spellbook/models/sudoswap/ethereum/sudoswap_ethereum_events.sql

```diff
@@ -0,0 +1,240 @@
+{{ config(
+        alias ='events',
+        materialized ='incremental',
+        file_format ='delta',
+        incremental_strategy='merge',
+        unique_key='unique_trade_id'
+        )
+}}
+
+--base table CTEs
+WITH 
+    swaps as (
+        SELECT 
+            * 
+        FROM (
+            SELECT 
+                contract_address
+                , call_tx_hash
+                , call_trace_address
+                , call_block_time
+                , call_block_number
+                , call_success
+                , tokenRecipient as call_from
+                , 'Sell' as trade_category 
+            FROM {{ source('sudo_amm_ethereum','LSSVMPair_general_call_swapNFTsForToken') }}
+            UNION ALL
+            SELECT 
+                contract_address
+                , call_tx_hash
+                , call_trace_address
+                , call_block_time
+                , call_block_number
+                , call_success
+                , nftRecipient as call_from
+                , 'Buy' as trade_category 
+            FROM {{ source('sudo_amm_ethereum','LSSVMPair_general_call_swapTokenForAnyNFTs') }}
+            UNION ALL
+            SELECT
+                contract_address
+                , call_tx_hash
+                , call_trace_address
+                , call_block_time
+                , call_block_number
+                , call_success
+                , nftRecipient as call_from
+                , 'Buy' as trade_category 
+            FROM {{ source('sudo_amm_ethereum','LSSVMPair_general_call_swapTokenForSpecificNFTs') }}
+        ) s
+        {% if is_incremental() %}
+        -- this filter will only be applied on an incremental run. We only want to update with new swaps.
+        AND s.call_block_time >= (select max(block_time) from {{ this }})
+        {% endif %}
+    ),
+
+    pairs_created as (
+        SELECT 
+            _nft as nftcontractaddress
+            , _initialNFTIDs as nft_ids
+            , _fee as initialfee
+            , _assetRecipient as asset_recip
+            , output_pair as pair_address
+            , call_block_time as block_time
+        FROM {{ source('sudo_amm_ethereum','LSSVMPairFactory_call_createPairETH') }}
+        WHERE call_success
+    ),
+
+    owner_fee_update as (
+        SELECT 
+            *
+        FROM {{ source('sudo_amm_ethereum','LSSVMPair_general_evt_FeeUpdate') }}
```

Comment:
i'd apply the incremental filter here, since you use the date in join downstream

Reply:
can't filter this one since this is just checking for most recent fee update - so if the last updated swap was after the fee update then we'd lose data here.